compiler/rustc_passes/src/reachable.rs

```diff
@@ -373,7 +373,11 @@ fn reachable_set<'tcx>(tcx: TyCtxt<'tcx>, (): ()) -> FxHashSet<LocalDefId> {
     //         If other crates link to us, they're going to expect to be able to
     //         use the lang items, so we need to be sure to mark them as
     //         exported.
-    reachable_context.worklist.extend(access_levels.map.keys());
+    reachable_context.worklist = access_levels
+        .iter()
+        .filter_map(|(&id, _)| if access_levels.is_some(id) { Some(id) } else { None })
```

Comment:
Same as in `dead.rs`.
```suggestion
        .filter_map(|(&id, effective_vis)| if effective_vis.is_public_at_level(AccessLevel::ReachableFromImplTrait) { Some(id) } else { None })
```